lib/src/workspace.rs

```diff
@@ -141,6 +147,11 @@ fn init_working_copy(
         .check_out(workspace_name.clone(), &repo.store().root_commit())?;
     let repo = tx.commit(format!("add workspace '{}'", workspace_name.as_symbol()))?;
 
+    workspace_store.add(
+        &workspace_name,
+        &dunce::canonicalize(workspace_root).context(workspace_root)?,
```

Comment:
Callers should have canonicalized path since they create new `Workspace` object.

```rust
workspace_store.add(workspace.workspace_name(), workspace.workspace_root())
```

Reply:
Can you please expand on this? I have been quite confused regarding this since your last comment. What callers? Where is `workspace` object coming from? And will the `workspace.workspace_root()` be always canonicalized? As mentioned before the `workspace` created through `add` command didn't have canonicalized `workspace_root`.